lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -557,6 +557,27 @@ def confirm_namespace_exists
       @logger.info("Namespace #{@namespace} found")
     end
 
+    # make sure to never prune the ejson-keys secret
+    def confirm_ejson_keys_not_prunable
+      raw_secret, st, err = nil
+      with_retries(2) do
```

Comment:
I thought we were going to refactor EjsonSecretProvisioner#run_kubectl_json to do this?

Reply:
I also forgot we had discussed something around this. Now that you bring it up though, the idea I remember is to expose a `EjsonSecretProvisioner#ejson_keys` method, which would replace L562-567 and make it so that we only ever fetch that secret once. `run_kubectl_json` is just a private kubectl wrapper.